panic: Architecture isn't supported: loong64

Debian has [announced](https://lists.debian.org/debian-devel-announce/2023/08/msg00000.html) loong64 as a new port. While Incus hasn't yet been uploaded to Debian, LXD  encounters a run-time panic on that architecture ([buildd log](https://buildd.debian.org/status/fetch.php?pkg=lxd&arch=loong64&ver=5.0.2-5&stamp=1696461135&raw=0)), and I would expect Incus to have the same issue:

```
panic: Architecture isn't supported: loong64

goroutine 1 [running]:
github.com/lxc/lxd/shared/version.getUserAgent()
	github.com/lxc/lxd/shared/version/useragent.go:22 +0x548
github.com/lxc/lxd/shared/version.init()
	github.com/lxc/lxd/shared/version/useragent.go:15 +0x28
```

I'm not sure if this will require more than just adding loong64 to `shared/osarch/architectures.go`.

(Also reported to LXD as https://github.com/canonical/lxd/issues/12387)

--- a/shared/osarch/architectures.go
+++ b/shared/osarch/architectures.go
@@ -20,6 +20,7 @@ const (
 	ARCH_64BIT_RISCV_LITTLE_ENDIAN   = 12
 	ARCH_32BIT_ARMV6_LITTLE_ENDIAN   = 13
 	ARCH_32BIT_ARMV8_LITTLE_ENDIAN   = 14
+	ARCH_64BIT_LOONGARCH             = 15
 )
 
 var architectureNames = map[int]string{
@@ -37,6 +38,7 @@ var architectureNames = map[int]string{
 	ARCH_64BIT_MIPS:                  "mips64",
 	ARCH_32BIT_RISCV_LITTLE_ENDIAN:   "riscv32",
 	ARCH_64BIT_RISCV_LITTLE_ENDIAN:   "riscv64",
+	ARCH_64BIT_LOONGARCH:             "loongarch64",
 }
 
 var architectureAliases = map[int][]string{
@@ -53,6 +55,7 @@ var architectureAliases = map[int][]string{
 	ARCH_64BIT_MIPS:                  {"mips64el", "mips64le"},
 	ARCH_32BIT_RISCV_LITTLE_ENDIAN:   {},
 	ARCH_64BIT_RISCV_LITTLE_ENDIAN:   {},
+	ARCH_64BIT_LOONGARCH:             {"loong64"},
 }
 
 var architecturePersonalities = map[int]string{
@@ -70,6 +73,7 @@ var architecturePersonalities = map[int]string{
 	ARCH_64BIT_MIPS:                  "linux64",
 	ARCH_32BIT_RISCV_LITTLE_ENDIAN:   "linux32",
 	ARCH_64BIT_RISCV_LITTLE_ENDIAN:   "linux64",
+	ARCH_64BIT_LOONGARCH:             "linux64",
 }
 
 var architectureSupportedPersonalities = map[int][]int{
@@ -87,6 +91,7 @@ var architectureSupportedPersonalities = map[int][]int{
 	ARCH_64BIT_MIPS:                  {ARCH_32BIT_MIPS},
 	ARCH_32BIT_RISCV_LITTLE_ENDIAN:   {},
 	ARCH_64BIT_RISCV_LITTLE_ENDIAN:   {},
+	ARCH_64BIT_LOONGARCH:             {},
 }
 
 const ArchitectureDefault = "x86_64"

